### ISSUE TITLE
Screen intermittently freezes

### ISSUE THREAD

**Issue by DemiMarie** [GrapheneOS/os-issue-tracker#1857]
On my phone (Pixel 6a) the screen intermittently freezes.  When this happens, touch input stops working, but audio does _not_ stop working: if I am in a call, the call keeps going.  Resolving the problem requires a hard reboot via the power button.

Edit: Rebooting the device _always_ fixes the problem, so I suspect this is a software problem (SurfaceFlinger crash?) and not a hardware problem.  This problem appeared after an update (sometime in November I think?); prior to that, I would get “System UI is not responding” instead.

**Comment by thestinger** [comment]
We're unable to reproduce this and we don't have other reports of it. If it's a software issue, it's likely closely tied to your configuration. You need to get logs after it happens if we'll have any hope of diagnosing it. You're likely going to need to debug what's wrong with your configuration. Start by disabling any accessibility services, device managers, custom launcher, etc.

**Comment by DemiMarie** [comment]
> You need to get logs after it happens if we'll have any hope of diagnosing it.

I will try to do this.  Are the logs persistent across a reboot?

> You're likely going to need to debug what's wrong with your configuration.

The only potentially unusual aspects I can think of are that I have sandboxed Google Play Services on my primary (and only) profile and have Google Services for VR present but disabled.  I saw that some system apps had some permissions listed as disabled and enabled them as I figured that could be the cause.

> Start by disabling any accessibility services, device managers, custom launcher, etc.

I do not have any of these running.

**Comment by thestinger** [comment]
> I will try to do this. Are the logs persistent across a reboot?

No, but it sounds like you should be able to get logs via `adb logcat` since the device remains working other than the screen freezing. It's possible it does take input and you just can't see it.

**Comment by thestinger** [comment]
> The only potentially unusual aspects I can think of are that I have sandboxed Google Play Services on my primary (and only) profile and have Google Services for VR present but disabled. I saw that some system apps had some permissions listed as disabled and enabled them as I figured that could be the cause.

You can reset app settings for all apps but you'll need to re-grant permissions, special access permissions (battery optimization exception for Play services, etc.) and so on. You'll also need to re-revoke Network and Sensors, although we have a default off option available for Sensors now.

**Comment by DemiMarie** [comment]
> > I will try to do this. Are the logs persistent across a reboot?
> 
> No, but it sounds like you should be able to get logs via `adb logcat` since the device remains working other than the screen freezing.

The bug triggered recently, but I needed to use the device right away so I had to reboot it.  I will see if it happens at a more convenient time.

> It's possible it does take input and you just can't see it.

I am pretty sure that it does not, since answering a call does not work.

> > The only potentially unusual aspects I can think of are that I have sandboxed Google Play Services on my primary (and only) profile and have Google Services for VR present but disabled. I saw that some system apps had some permissions listed as disabled and enabled them as I figured that could be the cause.
> 
> You can reset app settings for all apps but you'll need to re-grant permissions, special access permissions (battery optimization exception for Play services, etc.) and so on. You'll also need to re-revoke Network and Sensors, although we have a default off option available for Sensors now.

I will try to get logs from an instance of the bug first.

**Comment by thestinger** [comment]
It could be a hardware issue. It sounds like the OS is still working but touchscreen isn't working for either display or input which would be strange as a software issue since they're almost entirely separate things even at a driver level. Maybe a cable is just loose. The OS may not handle it disconnecting and reconnecting.

**Comment by DemiMarie** [comment]
> It could be a hardware issue. It sounds like the OS is still working but touchscreen isn't working for either display or input which would be strange as a software issue since they're almost entirely separate things even at a driver level. Maybe a cable is just loose. The OS may not handle it disconnecting and reconnecting.

On desktop Linux, both display and input go through the X server or Wayland compositor.  Could the problem be a SurfaceFlinger crash?

**Comment by thestinger** [comment]
It's a separate process and if the processes crash they get restarted quickly or trigger a soft reboot restarting most of the application layer userspace.

**Comment by SardarMark** [comment]
I am new to GrapheneOS. I  just had the same issue on my Pixel 7 (using for little over a month). Entire screen was frozen, I was not able to hard reboot or anything (Screen and Button wasn't responding). I can see the screen intermittently and saw the message "UI not responding" message. Hold the power button + volume up\down didn't help much as the GrapheneOS boot screen came and froze again. I have multiple profile setup. The profile I was using has Google Play Services installed and few other apps. My Main profile has nothing installed (No Google Stuff).
I got a bit nervous as no matter what I tried, the phone wasn't responding. I removed the SIM and kept on trying, took about 15 min. Eventually I was able to properly reboot, (Got to the screen - "warning a different OS is loading...") Now the phone is working as expected.  This is the first time I ran into this issue, I will try to see if I can enable debugging and get the log next time. 
It took me a little googling to find this GitHub post, it's possible that other users might do what I did and move on (with fingers crossed).

**Comment by mpiraz** [comment]
I've got  the same issue  since an upgrade after september/october. Its randomly ... sometimes with black screen sometimes with screen freezed . It occur four/five times a week (Pixel 6)

**Comment by DemiMarie** [comment]
> I've got the same issue since an upgrade after september/october. Its randomly ... sometimes with black screen sometimes with screen freezed . It occur four/five times a week (Pixel 6)

Thanks!  Pixel 6 or 6a?  Can you provide logs?

**Comment by mpiraz** [comment]
Pixel 6. How can I provide logs ? Can you help me with this? I guess logs
were completely deleted after reboot ...

**Comment by thestinger** [comment]
We need some kind of logs or additional information to act on this.

**Comment by mpiraz** [comment]
How extract logs ?

**Comment by thestinger** [comment]
Bug report capture tool in developer options is the best way for this. You can email it to us.

**Comment by DemiMarie** [comment]
@thestinger is it possible to use this if the screen is not responding?

**Comment by thestinger** [comment]
Not really but you can use `adb logcat -d > logs.txt` and also `adb shell dumpsys > dumpsys,txt`.

**Comment by thestinger** [comment]
There may be useful info in the bug report zip in the next boot though.

**Comment by DemiMarie** [comment]
Could GrapheneOS automatically store if the user holds the power button for long enough?  Firmware resets the device after 30 seconds, so 15 seconds should give the OS more than enough time.

**Comment by thestinger** [comment]
It takes longer than that to do a full bug report capture.

**Comment by DemiMarie** [comment]
I see.  What about some sort of hotkey for this, like power + vol up + vol down while shaking the device?

**Comment by mpiraz** [comment]
I've just sent a complete new bug report . It happened about  20 minutes ago.
Regards

**Comment by cjseaslug** [comment]
Have also been experiencing this on Pixel 5 with no concrete reproducing steps, but it's happening about 2x per day now. Has happened since quite a few versions ago, I would say approx 2 months and has gotten worse lately. Auto-update on.

Sorry I can't add any more info than "this happens to me" but it's the best I've got right now :(

**Comment by cjseaslug** [comment]
One update: can confirm screenshots and accessibility services (in my case, Key Mapper triggering flashlight from button press combo) while the picture is frozen on screen.

Screenshots display what was frozen on screen but seemingly in the wrong orientation (landscape, while i was holding phone portrait). Still no log, apologies.

**Comment by nifri** [comment]
I have the same issue on my Pixel 5 since about the same time that @DemiMarie mentioned. I haven't figured out how to reproduce it reliably. When I use a combination of Newpipe to watch a video from YT, a headset, pause a video, let the screen lock I have encountered this issue several times a day. But not every time. 

I will extract the logs that @thestinger mentioned. I'm concerned that they will contain personal information like my phone number or other device IDs. Is that the case? If yes I don't want to attach them here. Where could I send them?

**Comment by SapientHetero** [comment]
This happens on my Pixel 5a too. I've recently noticed lots of flakiness, such as delays of several seconds after touching the recent apps soft key before display changes. Sometimes I see 'No apps found' when the display does updates, despite running a variety of apps for hours before this happened.

**Comment by groetg** [comment]
I have a Pixel 6, also auto update and experience since Monday same trouble. I wil try capture a log, but don't know how to 'adb' stuff etc. It happened 4 times since Monday. Can't reproduce it.

**Comment by mpiraz** [comment]
@thestinger Have you received my email with bug report zip file ? The only way to extract bug report is after system reboot but I don't figure out if it works to solve the issue

**Comment by DemiMarie** [comment]
@mpiraz I believe you will need to use `adb logcat` while the system is frozen.

**Comment by groetg** [comment]
I logged out at the Google play store after the last incident en now for two days, no black screen...
(I logged in because I want to try a paid app Trenera for my running hobby)

**Comment by DemiMarie** [comment]
@groetg On GrapheneOS Google Play should not be able to do that, so that it can is itself a bug.

**Comment by DemiMarie** [comment]
> > @groetg On GrapheneOS Google Play should not be able to do that, so that it can is itself a bug.
> 
> What does "that" here refer to?

No unprivileged app should be able to make the screen unresponsive in a way that requires a reboot to recover from, even if it is malicious.

**Comment by Ashaman-** [comment]
I have encountered this, also on a Pixel 6a. I will enable ADB, and try to get logs the next time it haplens.

**Comment by alex8065** [comment]
Happening to me on a Pixel 5, will email a log capture next time I catch it.

**Comment by kaleresan** [comment]
Happening to me on a Pixel 4a. will try to get a log capture next time it happens

**Comment by mpiraz** [comment]
It has been happening  once a week (pixel 6) ... Waiting for a solution !
Thanks

**Comment by DemiMarie** [comment]
@mpiraz can you get a log capture when it does happen?

**Comment by mpiraz** [comment]
Yes I can, but only after reboot the system . I sent a couple of times the
Log capture ... Did you get it?

**Comment by DemiMarie** [comment]
@mpiraz can you get a log capture _before_ rebooting, using `adb logcat`?  You’ll need debugging enabled.

**Comment by mpiraz** [comment]
It's my business phone ... adb logcat requires a coder knowledge (I'm a
manager and lawyer). But i should try ... not by shell but with android
studio package starting from here ?
Is it correct ?
[image: image.png]
--------------------------
Maurizio Pirazzini

E-mail: ***@***.***
PEC (certified electronic mail): ***@***.***
***@***.***>
Mobile phone +39 347 4204350
http://www.linkedin.com/in/mauriziopirazzini




Prima di stampare, pensa all'ambiente ** Think about the environment before
printing.



Il giorno mer 8 mar 2023 alle ore 16:04 Demi Marie Obenour <
***@***.***> ha scritto:

**Comment by alex8065** [comment]
Does anyone know if it's possible to enable previous boot logs in GrapheneOS?

**Comment by davidhaley** [comment]
This used to happen to me; however, I discovered/started using the `Hold power -> End session` feature, from within a second user profile, and it seems to behave normally after that.

**Comment by soupslurpr** [comment]
This used to happen to me too, it has stopped.

**Comment by SapientHetero** [comment]
Still happening to me too. Sometimes I get 'System UI has stopped' message.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by SapientHetero** [comment]
I've noticed a pattern; this usually happens after copying text to the clipboard in an app or copying a link to a video in YouTube.

Sometimes it recovers on its own, other times I must reboot to restore normal operation. When time permits I'll look into grabbing the log file during one of these events.

**Comment by zWhdmB5T** [comment]
It's quite similar here…

* Device: Pixel 6a
* Build-Number: TQ3A.230805.001.2023090600 (but the bug is lasting for about half a year or so)

# Problem
Sometimes, the screen freezes. This happens not daily, but some times a week. It mostly occurs when using the browser [Mull — Privacy oriented web browser](https://f-droid.org/en/packages/us.spotco.fennec_dos/), but it's not strongly limited to that. Then, touch input is NOT working. But the phone is still active and alive.

# Work-around (that works for me)
1. Press Power button shortly to switch off display and lock device
2. Press Power button again to show un-lock screen
3. Use fingerprint login to un-lock
4. Continue regular use
5.  N.B.: NO reboot needed here

**Comment by artproject-2004** [comment]
From time to time I have the same issue (after update to Androud 14).

- GrapheneOS with Android 14.

**Comment by abbluiz** [comment]
I've had the same issue with Pixel 5a. It is really weird. Out of nowhere the system stops responding, the screen freezes and no inputs respond. After a few seconds, the screen goes black. Sometimes, it feels like the OS is still running, but recently the black screen looks more like the device shuts down.

To get it back to working, I sometimes can use a combination of keys (power key + volume up or down) to get it to boot again. However, it almost never work and it seems random. Sometimes it won't boot for hours.

**Comment by cjseaslug** [comment]
I disabled Neo Store and that fixed it on my 5a.  Disabling auto update also seems to fix it.

I can only hope it's the same for anyone else here but I have my doubts!

Nov 3, 2023 10:21:26 PM LABB ***@***.***>:

**Comment by zWhdmB5T** [comment]
> It's quite similar here…
Still having this issue…

> 
>     * Device: Pixel 6a
> 
>     * Build-Number: TQ3A.230805.001.2023090600 (but the bug is lasting for about half a year or so)
UP1A.231005.007.2023103100 now

> # Problem
> 
> Sometimes, the screen freezes. This happens not daily, but some times a week. It mostly occurs when using the browser [Mull — Privacy oriented web browser](https://f-droid.org/en/packages/us.spotco.fennec_dos/), but it's not strongly limited to that. Then, touch input is NOT working any more. But the phone is still active and alive.
> # Work-around (that works for me)
> 
>     1. Press Power button shortly to switch off display and lock device
> 
>     2. Press Power button again to show un-lock screen
> 
>     3. Use fingerprint login to un-lock
> 
>     4. Continue regular use
> 
>     5. N.B.: NO reboot needed here

Currently, not the entire screen is unusable for me. Right now, it seems that only the Android 3 buttons menu at the bottom is affected.
The work-around described above still holds capable.

**Comment by tropicalMood** [comment]
Seems like [issue](https://github.com/GrapheneOS/os-issue-tracker/issues/2646) I created should be same or simillar as you describing here

**Comment by DemiMarie** [comment]
Personally I have not had this problem for a while.  The upgrade to Android 14 might have fixed it for me.

**Comment by DemiMarie** [comment]
@zWhdmB5T you ran into a different problem, mind opening a different issue?

**Comment by thestinger** [comment]
Original issue is resolved There are new issues filed about gesture and launcher regressions in Android 14.

**Comment by zWhdmB5T** [comment]
> @zWhdmB5T you ran into a different problem, mind opening a different issue?

@DemiMarie yes, sounds good. Problems seem to be different. Initial issue seems to be resolved.

**Comment by zWhdmB5T** [comment]
> Original issue is resolved There are new issues filed about gesture and launcher regressions in Android 14.

@thestinger yes, sounds good. Problems seem to be different. Initial issue seems to be resolved.

**Comment by abbluiz** [comment]
Update: my phone died and won't turn back on. I guess it was one of those Pixel 5A hardware issues. The phone was 2 years old. Very disappointed and can't get Google support cause I live in Brazil where this phone is not officially sold.

**Comment by SapientHetero** [comment]
Mine did the same thing. I was using it and the screen suddenly went black. The phone's been unresponsive every since.

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by richardkv** [comment]
Same thing happening here on a Pixel 6a and 7pro. Intermittent screen freezing. Both phones have 2 users. The 7 pro has been perfect for about 3 days - no problems, then suddenly the lagging and freezing begins. Unable to swipe up out of a full screen app or swipe back. Turning the screen on and off usually will do the trick and allow me to (laggily) swipe up and out.

**Comment by richardkv** [comment]
Same thing happening here on a Pixel 6a and 7pro. Intermittent screen freezing. Both phones have 2 users. The 7 pro has been perfect for about 3 days - no problems, then suddenly the lagging and freezing begins. Unable to swipe up out of a full screen app or swipe back. Turning the screen on and off usually will do the trick and allow me to (laggily) swipe up and out.

**Comment by Hampirico** [comment]
Same thing happening here on a Pixel 6a and 7pro. Intermittent screen freezing. Both phones have 2 users. The 7 pro has been perfect for about 3 days - no problems, then suddenly the lagging and freezing begins. Unable to swipe up out of a full screen app or swipe back. Turning the screen on and off usually will do the trick and allow me to (laggily) swipe up and out.

**Comment by thestinger** [comment]
Please open a new issue.

**Comment by thestinger** [comment]
@abbluiz 

> Update: my phone died and won't turn back on. I guess it was one of those Pixel 5A hardware issues. The phone was 2 years old. Very disappointed and can't get Google support cause I live in Brazil where this phone is not officially sold.

That's a hardware issue.

**Comment by thestinger** [comment]
@Hampirico 

> Mine did the same thing. I was using it and the screen suddenly went black. The phone's been unresponsive every since.

Also a hardware issue if it's unresponsive before the OS is loaded.

**Comment by thestinger** [comment]
@Hampirico

This isn't the same as the issue filed here. You have the upstream Android 14 gesture / launcher related issue. There are separate issues filed about that. This issue was resolved.
